### ISSUE TITLE
Approve Release

### ISSUE THREAD

**Issue by jernejk** [panuoksala/streamdeck-azuredevops-plugin#4]
As a tech lead, I need to approve releases before they go into staging/production.

It would be great if I could see when a release is waiting for approval and then be able to approve it with a button press.

NOTE: I would need the same feature for builds as well but currently it doesn't seem to be possible to do it via APIs.

**Comment by panuoksala** [comment]
https://docs.microsoft.com/en-us/rest/api/azure/devops/release/Approvals/List?view=azure-devops-rest-6.0

**Comment by panuoksala** [comment]
Add new selection into Pipeline type: Approval.
Then we can query approval against given release pipeline id.

For example if the user starts release with id X, we can get the approval id for the release and use it to approve release. Problem is that release can have multiple approvals even in the same level of stages. So we can only approve like the first one(?)

We need to also implement icon for waiting for approval state.

**Comment by jernejk** [comment]
Agree, this feature needs a bit more thinking to get it done and I'm currently not working on a project that use approval for releases.

A solution for multiple approvals could be that single tap switches between different approvals and long press approves it.

Most pipelines I work on with clients, they merge build and release yaml into build pipeline. This makes approval hard because there is no API for build approvals. 🤔
